### PR TITLE
Add test to update io_profile in longevity

### PR DESCRIPTION
**Pull request by Santhoshairody**
Add test to update io_profile in longevity

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:
http://jen-endurance.pwx.dev.purestorage.com/me/my-views/view/all/job/Users/job/santhoshm/job/tp-longevity-volumeupdate-dev/88/consoleFull
I will add data integrity check when it is available.

### CODE REVIEW

File: tests/testTriggers.go [hunk 1]
```diff
@@ -4770,6 +4771,105 @@ func TriggerVolumeUpdate(contexts *[]*scheduler.Context, recordChan *chan *Event
...
+						log.Infof("Validating context: %v", ctx.App.Key)
+						ctx.SkipVolumeValidation = false
+						ValidateContext(ctx, &errorChan)
+						for err := range errorChan {
+							log.Infof("Error: %v", err)
+							UpdateOutcome(event, err)
+						}
+					}
+					if count < 1 {
+						log.InfoD("Wait for 15 minutes before restting back")
```
Comment (lsrinivas-pure) [inline tests/testTriggers.go:4812]: nit: spell check resetting

File: tests/testTriggers.go [hunk 2]
```diff
@@ -4770,6 +4771,105 @@ func TriggerVolumeUpdate(contexts *[]*scheduler.Context, recordChan *chan *Event
...
+	updateMetrics(*event)
+}
+
+// updateIOProfileOnVolumes this method is responsible for updating IO priority on Volumes.
+func updateIOProfileOnVolumes(contexts *[]*scheduler.Context, event *EventRecord) {
+	for _, ctx := range *contexts {
+		var appVolumes []*volume.Volume
+		var volumeSpec *apios.VolumeSpecUpdate
+		var err error
+		//var requiredIOProfile apios.VolumeSpecPolicy_IoProfile
```
Comment (lsrinivas-pure) [inline tests/testTriggers.go:4830]: delete if not required

File: tests/testTriggers.go [hunk 3]
```diff
@@ -4770,6 +4771,105 @@ func TriggerVolumeUpdate(contexts *[]*scheduler.Context, recordChan *chan *Event
...
+			log.InfoD("Volume: %s Current IO profile : %s, current derived IO profile %s", v.ID, currentIOProfile, derivedIOProfile)
+			if currentIOProfile == apios.IoProfile_IO_PROFILE_AUTO {
+				requiredIOProfile = apios.IoProfile_IO_PROFILE_NONE
+			} else if currentIOProfile == apios.IoProfile_IO_PROFILE_NONE {
+				requiredIOProfile = apios.IoProfile_IO_PROFILE_AUTO
+			}
+			log.InfoD("Expected IO Profile %v", requiredIOProfile)
+			volumeSpec = &apios.VolumeSpecUpdate{IoProfileOpt: &apios.VolumeSpecUpdate_IoProfile{IoProfile: requiredIOProfile}}
+			Inst().V.UpdateVolumeSpec(v, volumeSpec)
+			if err != nil {
```
Comment (lsrinivas-pure) [inline tests/testTriggers.go:4854]: no need for err check here UpdateOutcome will take care of it

File: tests/testTriggers.go [hunk 4]
```diff
@@ -4770,6 +4771,105 @@ func TriggerVolumeUpdate(contexts *[]*scheduler.Context, recordChan *chan *Event
...
+		for _, v := range appVolumes {
+			log.InfoD("Getting info from volume: %s", v.ID)
+			appVol, err := Inst().V.InspectVolume(v.ID)
+			if err != nil {
+				log.Errorf("Error inspecting volume: %v", err)
+			}
+			currentIOProfile := appVol.Spec.IoProfile
+			derivedIOProfile := appVol.DerivedIoProfile
+			log.InfoD("Volume: %s Current IO profile : %s, current derived IO profile %s", v.ID, currentIOProfile, derivedIOProfile)
+			if currentIOProfile == apios.IoProfile_IO_PROFILE_AUTO {
```
Comment (lsrinivas-pure) [inline tests/testTriggers.go:4846]: is profile journal not required?

Comment (Santhoshairody) [reply tests/testTriggers.go]: We will use AutoJournal, that should be enough. If required we will add it

**Comment by Santhoshairody** [comment]
Test log
http://jen-endurance.pwx.dev.purestorage.com/job/Users/job/santhoshm/job/tp-longevity-volumeupdate-dev/103/consoleFull

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Pull request merged**